lib/puppet/indirector/file_server.rb

```diff
@@ -29,6 +29,7 @@ def find(request)
     return nil unless path = mount.find(relative_path, request)
     result = model.new(path)
     result.links = request.options[:links] if request.options[:links]
+    result.checksum_type = request.options[:checksum_type] if request.options[:checksum_type]
```

Comment:
I see that too, I'll start looking at it.

On Fri, Jan 23, 2015 at 2:17 PM, Peter Huene <notifications@github.com>
wrote:

> In lib/puppet/indirector/file_server.rb
> <https://github.com/puppetlabs/puppet/pull/3507#discussion_r23484730>:
>
> > @@ -29,6 +29,7 @@ def find(request)
> >      return nil unless path = mount.find(relative_path, request)
> >      result = model.new(path)
> >      result.links = request.options[:links] if request.options[:links]
> > +    result.checksum_type = request.options[:checksum_type] if request.options[:checksum_type]
>
> After reviewing this fix and the fix for PUP-3888, would it be possible to
> set these options only once in a helper method, perhaps in
> Puppet::FileServing::TerminusHelper? Seem like there's four places where
> we're setting both links and checksum_type (the two in this PR, the one
> in the fix for PUP-3888, and one already present in
> Puppet::FileServing::TerminusHelper).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/3507/files#r23484730>.
>